cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzurePlatformResources.java

```diff
@@ -242,23 +242,23 @@ public CloudVmTypes virtualMachines(ExtendedCloudCredential cloudCredential, Reg
         VmType defaultVmType = null;
         Set<String> vmTypesWithoutResourceDisks = new HashSet<>();
         for (VirtualMachineSize virtualMachineSize : vmTypes) {
-            if (virtualMachineSize.resourceDiskSizeInMB() != NO_RESOURCE_DISK_ATTACHED_TO_INSTANCE) {
-                float memoryInGB = virtualMachineSize.memoryInMB() / NO_MB_PER_GB;
-                VmTypeMetaBuilder builder = VmTypeMetaBuilder.builder().withCpuAndMemory(virtualMachineSize.numberOfCores(), memoryInGB);
-
-                for (VolumeParameterType volumeParameterType : VolumeParameterType.values()) {
-                    if (volumeParameterType.in(MAGNETIC, SSD)) {
-                        volumeParameterType.buildForVmTypeMetaBuilder(builder, virtualMachineSize.maxDataDiskCount(), maxDiskSize);
-                    }
-                }
-
-                VmType vmType = VmType.vmTypeWithMeta(virtualMachineSize.name(), builder.create(), true);
-                types.add(vmType);
-                if (virtualMachineSize.name().equals(armVmDefault)) {
-                    defaultVmType = vmType;
+            float memoryInGB = virtualMachineSize.memoryInMB() / NO_MB_PER_GB;
+            VmTypeMetaBuilder builder = VmTypeMetaBuilder.builder().withCpuAndMemory(virtualMachineSize.numberOfCores(), memoryInGB);
+            for (VolumeParameterType volumeParameterType : VolumeParameterType.values()) {
+                if (volumeParameterType.in(MAGNETIC, SSD)) {
+                    volumeParameterType.buildForVmTypeMetaBuilder(builder, virtualMachineSize.maxDataDiskCount(), maxDiskSize);
                 }
+            }
+            if (virtualMachineSize.resourceDiskSizeInMB() != NO_RESOURCE_DISK_ATTACHED_TO_INSTANCE) {
```

Comment:
maybe better to add this value to the VmType and add the hasDiskRespurce method with this condition